### DISCUSSION TITLE
Default kernel prompt when retrieving data via a native plug-in (#Python)

### DISCUSSION THREAD

**Discussion by dcampillo** [microsoft/semantic-kernel#8084] [Q&A] [answered]
Hi everyone,

I have added 2 native plugin (Python) to a kernel. 
1 plugin is meant to retrieve content via vector search, the second one if retrieving contact information.

When invoking the kernel (kernel.invoke), both are chosen correctly when needed, BUT I don't know/understand what prompt is used to retrieve an answer from OpenAI.

The answer are good, but when asking for contact information i would prefer state that the answer should be a table and not a list of bullet points.

Thanks for your help.

David

**Comment by moonbox3** [comment, marked as answer]
Hi @dcampillo, I think there are a few ways of handling this.

1. You can use SK's prompt filter to either validate the prompt, or update it as needed. You can see a sample of this [here](https://github.com/microsoft/semantic-kernel/blob/main/python/samples/concepts/filtering/prompt_filters.py). 

You'll notice in this sample that we're update the prompt to include extra information:

```python
@kernel.filter(FilterTypes.PROMPT_RENDERING)
async def prompt_rendering_filter(context: PromptRenderContext, next):
    await next(context)
    context.rendered_prompt = f"You pretend to be Mosscap, but you are Papssom who is the opposite of Moscapp in every way {context.rendered_prompt or ''}"  # noqa: E501
```

2. You can add more information to the description of your `kernel_function` about how the model should respond. When you write a `kernel_function` decorator, you can include the `name` and the `description`. In the `description` you can include information to the model that for that particular function it should always respond with a table. When we create the JSON schema for that function (tool), we will include the description so the model will have some extra "context" to use while making a decision on what tool to call, and what information to include in the final chat completion response.
